### DISCUSSION TITLE
v2.10.0

### DISCUSSION THREAD

**Discussion by pi0** [nitrojs/nitro#2829] [Announcements]
📝 [Release Notes](https://github.com/nitrojs/nitro/releases/tag/v2.10.0)

[join discord](https://discord.nitro.build)

## ⭐ Highlights

- 🏠 New home
- 🛠️ Major internal overhaul
- 📅 Compatibility date
- 🗂️ Environment-specific handlers
- 📍 Route groups
- 📜 production mode OpenAPI
- 📦 Upgraded dependencies (crossws, db0 and jiti)
- 🎛️ Preset updates (cloudflare, netlify, vercel and more)

**Comment by johannschopplich** [comment]
Congratulations on this release! Feels more major than the minor version bump suggests. 🙌

**Comment by they-cloned-me** [comment]
Wow 🤯

**Comment by tobychidi** [comment]
@pi0  The `return type` signature for defineCacheFunction change to include undefined .i.e. `T | undefined`. Is that necessary? Its breaking my types at the moment.

**Reply by pi0** [reply]
Thnx! Should be fixed in nightly and next patch (https://github.com/nitrojs/nitro/pull/2848)

**Comment by adamdehaven** [comment]
Referencing [my comment here](https://github.com/nitrojs/nitro/pull/2831#issuecomment-2489000828) for anyone else landing here, if your build is broken as of `2.10.0` you'll likely need to update your `wrangler.toml` accordingly to support static assets:

```diff
# wrangler.toml

- [site]
- bucket = ".output/public"
+ assets = { directory = "./.output/public/", binding = "ASSETS" }
```
